src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,59 @@
 package core.basesyntax;
 
-class Pair {
+public class Pair<T, K> {
+    static final int CONSTANT = 31;
 
+    private T first;
+    private K second;
+
+    private Pair(T first, K second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <T, K> Pair of(T data, K data2) {
+        return new Pair<T, K>(data, data2);
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public K getSecond() {
+        return second;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+
+        if (null == object) {
+            return false;
+        }
+
+        if (this == object) {
+            return true;
+        }
+
+        if (object.getClass().equals(Pair.class)) {
+            Pair pair = (Pair) object;
```

Comment:
we should cat to our parametrized class(ignore the idea warnings)

Reply:
Done